src/components/common/FileUploadPdf.vue

```diff
@@ -0,0 +1,298 @@
+
+  /** Clears data and local state. */
+  public reset (): void {
+    // update parent
+    this.updateFile(null)
+    this.updateFileKey(null)
+
+    // clear local state
+    this.errorMessages = []
+
+    // force re-render to clear file input
+    this.count++
+  }
+
+  /** Resets input validation only. */
+  public resetValidation (): void {
+    this.errorMessages = []
+  }
+
+  /**
+   * "Shallow validates" this component.
+   * "Deep validation" is done when a file is selected.
+   */
+  public validate (): boolean {
+    // force re-render to clear file input
+    this.count++
+
+    // Logic is as follows:
+    // if (no file) and (required) => error
+    // if (no file) and (not required) => valid
+    // if (have file) and (have key) => valid
+    // if (have file) and (no key) => error
+
+    if (!this.file) {
+      if (this.isRequired) {
+        this.errorMessages = ['File is required']
+        return false
+      } else {
+        this.errorMessages = []
+        return true
+      }
+    }
+
+    // if we have a file key then file is valid and was uploaded
+    if (this.fileKey) return true
+
+    return false
+  }
+
+  /** When file is selected or cleared, validates the file and uploads it. */
+  protected async onChange (file: File): Promise<void> {
+    // update parent for later reactivity
+    this.updateFile(file)
+    this.updateFileKey(null)
+
+    // Logic is as follows:
+    // if (no file) and (required) => error
+    // if (no file) and (not required) => valid
+    // if (have file) and (required) => depends on file validation
+    // if (have file) and (not required) => depends on file validation
+
+    if (!file) {
+      if (this.isRequired) {
+        this.errorMessages = ['File is required']
+        return
+      } else {
+        this.errorMessages = []
+        return
+      }
+    }
+
+    // validate the file
+    this.errorMessages = ['Processing...']
+    const validFile = await this.validateFile(file)
+    if (!validFile) return
+
+    // upload the file
+    this.errorMessages = ['Uploading...']
+    const fileKey = await this.uploadFile(file)
+    if (!fileKey) return
+
+    // update parent
+    this.updateFile(file)
+    this.updateFileKey(fileKey)
+
+    // if we get this far then everything succeeded
+    this.errorMessages = []
+  }
+
+  /**
+   * Validates the PDF file. Note that some of this is async.
+   * @param file the file to validate
+   * @returns whether file is valid
+   */
+  private async validateFile (file: File): Promise<boolean> {
+    if (typeof file.arrayBuffer === 'undefined') return true
+
+    // verify file size
+    if (this.maxSize && (file?.size > (this.maxSize * 1024 * 1024))) {
+      this.errorMessages = [`Exceeds maximum ${this.maxSize} MB file size`]
+      return false
+    }
+
+    // try to retrieve file info
+    const fileInfo = await this.retrieveFileInfo(file).catch(() => null as PdfInfoIF)
+    if (fileInfo === null) {
```

Comment:
New: check for valid PDF.

(Also on like 174 below, which can fail for the same reason.)